qp_shotgun/kneaddata/kneaddata.py

```diff
@@ -187,37 +189,61 @@ def _run_commands(qclient, job_id, commands, msg):
     return True, ""
 
 
-def _per_sample_ainfo(out_dir, samples):
-    pf = []
+def _gzip_file(path):
+    with open(path) as in_file:
+        gz_path = '%s.gz' % path
+        with gopen(gz_path, "wb") as out_file:
+            out_file.writelines(in_file)
+
+    return gz_path
+
+
+def _per_sample_ainfo(out_dir, samples, fwd_and_rev=False):
     ainfo = []
-    for rp, _, _, _ in samples:
-        smd = partial(join, out_dir, rp)
-
-        # matching forward/reverse
-        fname = smd('%s.trimmed.1.fastq' % rp)
-        if exists(fname):
-            pf.append((fname, 'preprocessed_fastq'))
-        fname = smd('%s.trimmed.2.fastq' % rp)
-        if exists(fname):
-            pf.append((fname, 'preprocessed_fastq'))
+
+    if fwd_and_rev:
+        paired, ur1, ur2 = [], [], []
+        for rp, _, _, _ in samples:
+            smd = partial(join, out_dir, rp)
+
+            # matching forward/reverse
+            fname = smd('%s_paired_1.fastq' % rp)
+            if not exists(fname):
+                utime(fname)
+            paired.append((_gzip_file(fname), 'preprocessed_fastq'))
+            fname = smd('%s_paired_2.fastq' % rp)
+            if not exists(fname):
+                utime(fname)
+            paired.append((_gzip_file(fname), 'preprocessed_fastq'))
```

Comment:
how is a consumer of this artifact going to differentiate between the forward and reverse? based on filename? I have the feeling that it is a big soft that. This links back to @tanaes on gchat